### PR TITLE
Calculator project - Breakfast through dinner

### PR DESCRIPTION
**Pull request by corinnepingul**
This project satisfies all the requirements of breakfast, lunch, and dinner. I also included functionality for the program to return an integer or float (dessert requirement). I think now that we have learned about methods I would have definitely done this program with methods for code that is repeated.

### CODE REVIEW

**Comment by jnf** [comment]
Starting review!

File: calculator.rb [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  POWERS --> type \"exp\" or \"^\"
+  """
+# QUESTIONS----------------------------------------------------------------------
+
+# Asks user what type of math they'd like to do
+print "What type of calculation would you like to do? "
+operation = gets.chomp.strip
+
+# Prints error message if user types something that's not an accepted operation
+while (operation != "add") && (operation != "+") && (operation != "sub") && (operation != "-") && (operation != "mult") && (operation != "*") && (operation != "div") && (operation != "/") && (operation != "exp") && (operation != "^")
```
Comment (jnf) [inline calculator.rb:18]: Even though this is a really long line, your (quite admirable) use of parens really helps keep it readable. :)

I dig the use of the `while` loop for verifying the user input. Good call there too. Moving forward, the Array `#include?` method will help you express this same thought in a more succinct way.

File: calculator.rb [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+  print "That is not an available command. Please enter a correct operation: "
+  operation = gets.chomp.strip
+end
+
+# Gets first number from user (string)
+print "What is your first number: "
+first_num = gets.chomp
+
+# Prints error message if not a number
+while first_num != "0" && first_num.to_i == 0
```
Comment (jnf) [inline calculator.rb:28]: Figuring out how to do this validation in class was the highlight of my day. It definitely could use a clarifying comment though, for folks who weren't in the classroom with us on that glorious afternoon. :)

File: calculator.rb [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+# ADDITION
+if (operation == "add") || (operation == "+")
+  puts "#{ first_num } + #{ second_num } = #{ first_num + second_num }"
+
+# SUBTRATION
+elsif (operation == "sub") || (operation == "-")
+
+  # Asks to confirm order
+  print "Would you like to do: #{ first_num } - #{ second_num } (type \"1\") or #{ second_num } - #{ first_num } (type\"2\")? "
```
Comment (jnf) [inline calculator.rb:70]: This is a cool feature; I haven't seen this in anyone else' calculator.

File: calculator.rb [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+  end
+
+  # Does calculation in correct order
+  # Also checks to make sure user isn't trying to divide by zero
+  # Also gives remainder
+  if (order_of_oper == 1)
+    if (second_num == 0)
+      puts "You cannot divide by zero. Sorry!"
+    else
+      if (first_num.class == Fixnum)
```
Comment (jnf) [inline calculator.rb:110]: I love the check on this line; Ruby's math varies by object class, and controlling for that is fantastic.

In the current setup, if `second_num` is a Float (2.5), but `first_num` is a Fixnum (2), you'll end up with something like `2 / 2.5 = 0.8 with a remainder of 2.0`.  My advice would be to try checking for the reverse state and checking both variables, like `if (first_num.class == Float || second_num.class == Float)`.

**Comment by jnf** [comment]
You did a great job. The additional feature of asking for operation order is superb. My only suggestions are to add additional comments where intent may not be immediately clear and to be sure to test all the important pieces in your conditionals. Overall, though, really great job.

Do you have any questions about the project or my review comments?

File: calculator.rb (hunk 4 above)
Comment (corinnepingul) [inline calculator.rb:110]: Oh, I see! That's a good catch that I didn't think of.

**Comment by corinnepingul** [comment]
You're comments were awesome! Thanks! I'm still trying to figure out what types of comments to leave in my code (how much detail vs. not distracting from the code itself), so if you have any additional advice about that I welcome it :)

File: calculator.rb (hunk 2 above)
Comment (corinnepingul) [inline calculator.rb:28]: Ok, good point!

File: calculator.rb (hunk 3 above)
Comment (corinnepingul) [inline calculator.rb:70]: :) I think I actually got this idea from someone else!

**Pull request merged**